### PR TITLE
Changelog for Rust 1.69 :cherry_blossom:

### PR DESCRIPTION
**Pull request by xFrednet**
Roses are blue,
Violets are red,
Something is wrong,
We still commit!

---

changelog: none

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,11 +6,124 @@ document.
...
+
+### Suggestion Fixes/Improvements
+
+* [`box_default`]: The suggestion now includes the type for trait objects, when needed
+  [#10382](https://github.com/rust-lang/rust-clippy/pull/10382)
+* [`cast_possible_truncation`]: Now suggests using `try_from` or allowing the lint
+  [#10038](https://github.com/rust-lang/rust-clippy/pull/10038)
+* [`invalid_regex`]: Regex errors for non-literals or regular strings containing escape sequences will
+  now show the complete error
+  [#10231](https://github.com/rust-lang/rust-clippy/pull/10231)
```
[the remark this answers is not in the document]
Comment (xFrednet) [reply CHANGELOG.md:94]: The description is above the link. The normal order is:

```
* [`lint`] Description
  [#XYZ](PR-Link)
```

**Comment by xFrednet** [comment]
I forgot to include the backported PRs, now everything should be covered :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,11 +6,126 @@ document.
...
+### Moves and Deprecations
+
+* Moved [`uninlined_format_args`] to `pedantic` (Now allow-by-default)
+  [#10265](https://github.com/rust-lang/rust-clippy/pull/10265)
+* Moved [`unchecked_duration_subtraction`] to `pedantic` (Now allow-by-default)
+  [#10194](https://github.com/rust-lang/rust-clippy/pull/10194)
+
+### Enhancements
+
+* [`arithmetic_side_effects`]: No longer lints, if safe constant values are used.
```
Comment (llogiq) [inline CHANGELOG.md:47]:
```suggestion
* [`arithmetic_side_effects`]: No longer lints if safe constant values are used.
```
Stray comma

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,11 +6,126 @@ document.
...
+  on items visible within the current crate. For example, `pub(crate)` items.
+  [#10303](https://github.com/rust-lang/rust-clippy/pull/10303)
+* [`almost_swapped`]: Now detects almost swaps using `let` statements
+  [#10177](https://github.com/rust-lang/rust-clippy/pull/10177)
+* [`wildcard_enum_match_arm`]: Now lints missing private variants, for local enums
+  [#10250](https://github.com/rust-lang/rust-clippy/pull/10250)
+
+### False Positive Fixes
+
+* [`explicit_auto_deref`]: Now considers projections, when determining if auto deref is applicable
```
Comment (llogiq) [inline CHANGELOG.md:63]:
```suggestion
* [`explicit_auto_deref`]: Now considers projections when determining if auto deref is applicable
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,11 +6,126 @@ document.
...
+  [#10386](https://github.com/rust-lang/rust-clippy/pull/10386)
+* [`manual_let_else`]: Now considers side effects of branches, before linting
```
Comment (llogiq) [inline CHANGELOG.md:65]:
```suggestion
* [`manual_let_else`]: Now considers side effects of branches before linting 
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,11 +6,126 @@ document.
...
+
+### False Positive Fixes
+
+* [`explicit_auto_deref`]: Now considers projections, when determining if auto deref is applicable
+  [#10386](https://github.com/rust-lang/rust-clippy/pull/10386)
+* [`manual_let_else`]: Now considers side effects of branches, before linting 
+  [#10336](https://github.com/rust-lang/rust-clippy/pull/10336)
+* [`uninlined_format_args`]: No longer lints for arguments with generic parameters
+  [#10343](https://github.com/rust-lang/rust-clippy/pull/10343)
+* [`needless_lifetimes`]: No longer lints signatures in macros, if the lifetime is a metavariable
```
Comment (llogiq) [inline CHANGELOG.md:69]:
```suggestion
* [`needless_lifetimes`]: No longer lints signatures in macros if the lifetime is a metavariable
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,11 +6,126 @@ document.
...
+  [#10380](https://github.com/rust-lang/rust-clippy/pull/10380)
+* [`len_without_is_empty`]: No longer lints, if `len` as a non-default signature
```
Comment (llogiq) [inline CHANGELOG.md:71]:
```suggestion
* [`len_without_is_empty`]: No longer lints if `len` as a non-default signature
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -6,11 +6,126 @@ document.
...
+  [#10255](https://github.com/rust-lang/rust-clippy/pull/10255)
+* [`unusual_byte_groupings`]: Relaxed the required restrictions for specific sizes, to reduce false
```
Comment (llogiq) [inline CHANGELOG.md:73]:
```suggestion
* [`unusual_byte_groupings`]: Relaxed the required restrictions for specific sizes to reduce false
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -6,11 +6,126 @@ document.
...
+  positives
+  [#10353](https://github.com/rust-lang/rust-clippy/pull/10353)
+* [`manual_let_else`]: No longer lints `if-else` blocks if they can divergent
+  [#10332](https://github.com/rust-lang/rust-clippy/pull/10332)
+* [`expect_used`], [`unwrap_used`], [`dbg_macro`], [`print_stdout`], [`print_stderr`]: No longer lint
+  in test functions, if `allow-expect-in-tests` is set
```
Comment (llogiq) [inline CHANGELOG.md:79]:
```suggestion
  in test functions if `allow-expect-in-tests` is set
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -6,11 +6,126 @@ document.
...
+  [#10391](https://github.com/rust-lang/rust-clippy/pull/10391)
+* [`unnecessary_safety_comment`]: No longer lints code inside macros
+  [#10106](https://github.com/rust-lang/rust-clippy/pull/10106)
+* [`never_loop`]: No longer lints, for statements following break statements for outer blocks.
```
Comment (llogiq) [inline CHANGELOG.md:83]:
```suggestion
* [`never_loop`]: No longer lints statements following break statements for outer blocks.
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -6,11 +6,126 @@ document.
...
+  [#10311](https://github.com/rust-lang/rust-clippy/pull/10311)
+
+### Suggestion Fixes/Improvements
+
+* [`box_default`]: The suggestion now includes the type for trait objects, when needed
```
Comment (llogiq) [inline CHANGELOG.md:88]:
```suggestion
* [`box_default`]: The suggestion now includes the type for trait objects when needed
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -6,11 +6,126 @@ document.
...
+  [#10382](https://github.com/rust-lang/rust-clippy/pull/10382)
+* [`cast_possible_truncation`]: Now suggests using `try_from` or allowing the lint
+  [#10038](https://github.com/rust-lang/rust-clippy/pull/10038)
+* [`invalid_regex`]: Regex errors for non-literals or regular strings containing escape sequences will
+  now show the complete error
+  [#10231](https://github.com/rust-lang/rust-clippy/pull/10231)
+* [`transmutes_expressible_as_ptr_casts`]: The suggestion now works, if the base type is borrowed
```
Comment (llogiq) [inline CHANGELOG.md:95]:
```suggestion
* [`transmutes_expressible_as_ptr_casts`]: The suggestion now works if the base type is borrowed
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -6,11 +6,126 @@ document.
...
+* [`needless_return`]: The suggestion now works on if sequences
+  [#10345](https://github.com/rust-lang/rust-clippy/pull/10345)
+* [`needless_lifetimes`]: The suggestion is now machine applicable
+  [#10222](https://github.com/rust-lang/rust-clippy/pull/10222)
+* [`map_entry`]: The suggestion no longer expands macros
+  [#10346](https://github.com/rust-lang/rust-clippy/pull/10346)
+
+### ICE Fixes
+
+* [`needless_pass_by_value`]: Fixed an ICE, caused by how late bounds were handled
```
Comment (llogiq) [inline CHANGELOG.md:116]:
```suggestion
* [`needless_pass_by_value`]: Fixed an ICE caused by how late bounds were handled
```

**Review by llogiq — APPROVED** [review]
I found a few stray commas, with those fixed, r=me.

**Pull request merged**